src/subapps/dataExplorer/ColumnsSelector.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useEffect, useReducer, useState } from 'react';
+import { Input, Select, Col } from 'antd';
+import { RowRenderer } from '../../shared/molecules/TypeSelector/TypeSelector';
+import Light from '../../shared/components/Icons/Light';
+import { orderBy } from 'lodash';
+
+export type TColumn = { value: string; selected: boolean; key: string };
+type TColumnsSelectorProps = {
+  columns: TColumn[];
+  newItemsCount: number;
+  showNewItemsMessage: boolean;
+  loading: boolean;
+  onColumnSelect: (
+    e: React.MouseEvent<HTMLElement, MouseEvent>,
+    value: TColumn
+  ) => void;
+};
+type TColumnsSelectorState = {
+  inputColumns: TColumn[];
+  queryTerm: string;
+};
+
+const ColumnItem = ({ value }: TColumn) => {
+  return (
+    <Col span={20}>
+      <span title={value}>{value}</span>
+    </Col>
+  );
+};
+
+const ColumnsSelector = ({
+  columns,
+  newItemsCount,
+  showNewItemsMessage,
+  loading,
+  onColumnSelect,
+}: TColumnsSelectorProps) => {
+  const [{ inputColumns, queryTerm }, updateSearchResults] = useReducer(
+    (
+      previous: TColumnsSelectorState,
+      next: Partial<TColumnsSelectorState>
+    ) => ({
+      ...previous,
+      ...next,
+    }),
+    {
+      inputColumns: columns,
+      queryTerm: '',
+    }
+  );
+  const onSearchValueChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (e.type === 'click' || e.target.value === '') {
+      updateSearchResults({ inputColumns: columns, queryTerm: '' });
+    }
+    updateSearchResults({
```

Comment:
I think we should add a `return;` in the `if` condition above. Otherwise the `updateSearchResults` will be called twice if the condition is true.


Also, I'm not sure I understand the purpose of `if` block here. Why do we want to set `inputColumns` as all the `columns`? Shouldn't `inputColumns` be set to only those columns where the value matches the value of the clicked item?